pkg/builder/quarkus.go

```diff
@@ -208,6 +208,48 @@ func GenerateQuarkusProjectCommon(runtimeVersion string, quarkusVersion string,
 				},
 			},
 		},
+		maven.Plugin{
+			GroupID:    "com.google.cloud.tools",
+			ArtifactID: "jib-maven-plugin",
+			Version:    "3.3.1",
+			Dependencies: []maven.Dependency{
+				{
+					GroupID:    "com.google.cloud.tools",
+					ArtifactID: "jib-layer-filter-extension-maven",
+					Version:    "0.3.0",
+				},
+			},
+			Configuration: v1.PluginConfiguration{
+				Container: v1.Container{
+					Entrypoint: "INHERIT",
+					Args: v1.Args{
+						Arg: "jshell",
+					},
+				},
+				AllowInsecureRegistries: "true",
+				ExtraDirectories: v1.ExtraDirectories{
+					Paths: v1.Paths{
+						Path: v1.Path{
+							From: "../context",
+							Into: "/deployments",
+						},
+					},
+				},
+				PluginExtensions: v1.PluginExtensions{
+					PluginExtension: v1.PluginExtension{
+						Implementation: "com.google.cloud.tools.jib.maven.extension.layerfilter.JibLayerFilterExtension",
+						Configuration: v1.PluginExtensionConfiguration{
+							Implementation: "com.google.cloud.tools.jib.maven.extension.layerfilter.Configuration",
+							Filters: v1.Filters{
+								Filter: v1.Filter{
+									Glob: "/app/**",
```

Comment:
this creates a single layer right ? 

One of the nice things about the quarkus jib extensions is that it automatically configure jib to have some layers according to the app layout the quarkus builder generates

Reply:
In short yes we have a single layer created by the configuration: 
```
				ExtraDirectories: v1.ExtraDirectories{
					Paths: v1.Paths{
						Path: v1.Path{
							From: "../context",
							Into: "/deployments",
						},
					},
				},
```
The Filter code remove the default layers created by the google jib plugin.

I agree with you that the quarkus jib extension creates good layers according to the quarkus build, but sadly it can't make any incremental build.

I will see if I can make a better set of layers with the `ExtraDirectories` plugin configuration, that is a good idea.